exp/internal/controllers/machinepool_controller_test.go

```diff
@@ -336,82 +342,141 @@ func TestReconcileMachinePoolRequest(t *testing.T) {
 				Spec: expv1.MachinePoolSpec{
-					ClusterName:    "test-cluster",
-					ProviderIDList: []string{"test://id-1"},
-					Replicas:       ptr.To[int32](1),
+					ClusterName: "test-cluster",
+					Replicas:    ptr.To[int32](1),
 					Template: clusterv1.MachineTemplateSpec{
 						Spec: clusterv1.MachineSpec{
 							InfrastructureRef: corev1.ObjectReference{
 								APIVersion: builder.InfrastructureGroupVersion.String(),
 								Kind:       builder.TestInfrastructureMachineTemplateKind,
 								Name:       "infra-config1",
 							},
-							Bootstrap: clusterv1.Bootstrap{DataSecretName: ptr.To("data")},
+							Bootstrap:           clusterv1.Bootstrap{DataSecretName: ptr.To("data")},
+							NodeDeletionTimeout: &metav1.Duration{Duration: 1 * time.Minute},
 						},
 					},
+					ProviderIDList: []string{"aws:///us-test-2a/i-013ab00756982217f"},
 				},
 				Status: expv1.MachinePoolStatus{
-					Replicas:      1,
-					ReadyReplicas: 1,
 					NodeRefs: []corev1.ObjectReference{
-						{Name: "test"},
+						{
+							APIVersion: "v1",
+							Kind:       "Node",
+							Name:       "test-node",
+						},
 					},
-					ObservedGeneration: 1,
 				},
 			},
 			expected: expected{
-				result: reconcile.Result{},
-				err:    false,
+				mpExist: true,
+				result:  reconcile.Result{},
+				err:     false,
 			},
+			withTracker: true,
 		},
 		{
+			name: "Delete machinePool with nodeDeleteTimeout passed",
 			machinePool: expv1.MachinePool{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      "deleted",
+					Name:      "deleted-timeoutNode",
 					Namespace: metav1.NamespaceDefault,
 					Labels: map[string]string{
 						clusterv1.MachineControlPlaneLabel: "",
 					},
 					Finalizers:        []string{expv1.MachinePoolFinalizer},
-					DeletionTimestamp: &time,
+					CreationTimestamp: metav1.Time{Time: timeNow.Add(time.Minute * -2)},
+					DeletionTimestamp: &metav1.Time{Time: timeNow.Add(time.Minute * -1)},
 				},
 				Spec: expv1.MachinePoolSpec{
 					ClusterName: "test-cluster",
 					Replicas:    ptr.To[int32](1),
 					Template: clusterv1.MachineTemplateSpec{
 						Spec: clusterv1.MachineSpec{
-							InfrastructureRef: corev1.ObjectReference{
-								APIVersion: builder.InfrastructureGroupVersion.String(),
-								Kind:       builder.TestInfrastructureMachineTemplateKind,
-								Name:       "infra-config1",
-							},
-							Bootstrap: clusterv1.Bootstrap{DataSecretName: ptr.To("data")},
+							NodeDeletionTimeout: &metav1.Duration{Duration: 10 * time.Second},
+						},
+					},
+					ProviderIDList: []string{"aws:///us-test-2a/i-013ab00756982217f"},
+				},
+				Status: expv1.MachinePoolStatus{
+					NodeRefs: []corev1.ObjectReference{
+						{
+							APIVersion: "v1",
+							Kind:       "Node",
+							Name:       "test-node",
 						},
 					},
 				},
 			},
 			expected: expected{
-				result: reconcile.Result{},
-				err:    false,
+				mpExist: false,
+				result:  reconcile.Result{},
+				err:     true,
```

Comment:
```suggestion
				// Note: We expect an error here because patchHelper won't be able to patch the MachinePool after the finalizer was removed (and the MachinePool object is then gone)
				// The important part is that we verify that the MachinePool doesn't exist anymore
				err:     true,
```